text/0000-javascript-module-api.md

```diff
@@ -0,0 +1,1058 @@
+and name them impacts new learners and seasoned veterans alike. Thus we must try
+to find a balance between predictability for new and intermediate users, and
+terseness for experienced developers with large apps.
+
+There is another goal at play: we would like to help dispel the misconception
+that Ember is a monolithic framework. Ideally, module names help us tell a story
+about Ember's layered features. Rather than inheriting the entire framework at
+once, you can pull in just the pieces you need.
+
+For that reason, package names should assist the developer in understanding what
+capabilities are added by bringing in that new package. We should pick
+meaningful names, not let our public API be a by-product of how Ember's
+internals are organized.
+
+A full table of proposed mappings from global to module is available in
+[Addendum 1 - Table of Module Names and Exports by
+Global](#addendum-1---table-of-module-names-and-exports-by-global) and [Addendum
+2 - Table of Module Names and Exports by
+Package](#addendum-2---table-of-module-names-and-exports-by-package). Because
+there is some implicit functionality that you get when loading Ember that is not
+encapsulated in a global property (for example, automatically adding prototype
+extensions), there is also [Addendum 3 - Table of Modules with Side
+Effects](#addendum-3---table-of-modules-with-side-effects).
+
+Before diving in to these tables, however, it may be helpful to understand some
+of the thinking that guided this proposal.
+
+### Use Scoped Packages
+
+Last year, [npm introduced support for scoped packages][scoped-packages]. Scopes
+are similar to organizations on GitHub. They allow us to use any package name,
+even if it's already in use on npm, by namespacing it inside a scope.
+
+[scoped-packages]: http://blog.npmjs.org/post/116936804365/solving-npms-hard-problem-naming-packages
+
+For example, the [`component`][component] package is already reserved by an
+unmaintained tool; we couldn't use `component` as a package name even if we
+wanted to.
+
+[component]: https://www.npmjs.com/package/component
+
+However, scopes allow us to create a package named `component` that lives under
+the `@ember` scope: `import Component from "@ember/component"`.
+
+The advantages of using scoped packages, as this proposal does, are two-fold:
+
+1. "Official" packages are clearly differentiated from community packages.
+2. There is no risk of naming conflicts with existing community packages.
+
+Note that actually publishing packages to npm may not be immediately necessary
+to implement this RFC. We should still design around this constraint so that we
+have the option available to us in the future. For more discussion, see the
+[Distribution unresolved question](#distribution).
+
+### Prefer Common Terminology
+
+Module names should use terms people are more likely to be familiar with. For
+example, instead of the ambiguous `platform`, polyfills should be in a module
+called `polyfill`.
```

Comment:
I think a big benefit here is that this reorganization could increase the velocity of contributions back to the framework. I've been working with Ember for 4+ years and I still get lost trying to find where to make bug fixes in the source code. An easier to navigate project should help address this